### ISSUE TITLE
build target selection button

### ISSUE THREAD

**Issue by nholthaus** [vector-of-bool/vscode-cmake-tools#30]
A lot of CMake builds use a "superbuild" methodology, in which the `ALL` build consists of many individual targets. Often when developing such a project, you only want to build the single target you are actively working on to save time.

This can be done through the `CTRL+SHFT+P` menu currently, but it's clunky. I recommend:
- Adding another button similar to the `Release` button, which would allow the user to input the desired target
- The `Build` button would then show `Build ([target_name])` instead of `Build (all)`
- The target name would be propagated to the `--target` argument of the CMake command line.

Additionally, it might be nice to add a setting called `cmake.initialTarget` which would have a similar function to `cmake.InitialBuildType`, but for the target setting.

**Comment by vector-of-bool** [comment]
There is a command ``CMake: Set the default build target`` (Key is ``cmake.setDefaultTarget``) which does modify the default target when invoking ``cmake.build`` (As well as change the status bar button). Does this suffice? Automatically asking CMake for a list of available targets is a little bit trickier, but can be done. I thought about doing it earlier but never got around to doing it.

**Comment by nholthaus** [comment]
It's definitely usable as is, once I found those commands. As a new visual studio code user (who wasn't familiar with the command palette), I was using CMake tools for like 3 weeks without knowing it could even build code, so I still think the button would be helpful from a UX perspective. An enumerated target list like you mentioned would also be really nice, because in a large project like mine with 120+ targets, it's hard to remember them all.

I think my problem (which may be common to a lot of users) is that I did a search for 'cmake' in the code extension menu, and figured I was done, without realizing that it was actually pretty important to click on the product links and read the fine print about how each extension works. For people like me, the button would at least clue you in to the fact that CMake Tools has those capabilities.
